Layout  main page lection 3

lk4n opened the pull request:
Вариант №1
Страница компонента на гридах с блоком container

lk4n committed:
Init

lk4n committed:
Config changes

postcss-scss works as parser
added more options for scripts

lk4n committed:
Add support of minified main.css

lk4n committed:
Move to scss syntax

Changed syntax to scss for components which ready for now

lk4n committed:
Package and postcss config update

lk4n committed:
Merge branch 'master' into tools

lk4n committed:
Fix card block size in flexbox

lk4n committed:
Add stylelintignore

Ignoring minified css files in stylelineignore

lk4n committed:
Using prettier and stylelint

Set indentation to 2 spaces for prettier and stylelint
Set end on line in files for styelint rules because of prettier

lk4n committed:
Update header

Header gets size according to its content

lk4n committed:
Update footer for both main and component pages

lk4n committed:
Working on component page

Added component template.html
Added new blocks logo, sidebar, menu, link
Fix in footer block
Aded vars for new blocks
Fix in black color typo

lk4n committed:
Improving sidebar

After layout analisis condensed menu and link blocks into sidebar because of unnecessary complexity and low chance of reuse

lk4n committed:
Tools: simplified PostCSS plugin setup

Disabled PreCSS
Added plugin 'postcss-advanced-variables'
Added plugin 'postcss-at-rules-variables'

lk4n committed:
Sidebar names resolve

lk4n committed:
Component page

Merge logo block to header

Improve header block:
  - add header navigation
  - add header theme switch placeholder
Add page_flex modifier for flex realisation
Fixed typos in _vars.scss, _header.scss, _footer.scss

lk4n committed:
Navigation refactoring

New menu block, again
Menu using inheritance to handle colors

lk4n committed:
Improve scss support

Added Sass font variables
Added Sass mixins and first mixin for fonts
Fixed some typos

lk4n committed:
Fix preview

lk4n committed:
New block component

lk4n committed:
Doing grids....

lk4n committed:
Fix macro layout for component theme

Component page gets theme modifier

lk4n committed:
Component typos fix

lk4n committed:
Extend header section of the site

Add logo block
Add logo, menu for both main and component pages

lk4n committed:
Update

Add sidebar and component width to grid container .page__inner
Set background for header and footer instead of transparent to fix sliding during for small page widths

jsru commented:
Добавляю преподавателя (@GSemikozov) для код-ревью.

GSemikozov commented on components/card/_card.scss:
```diff
@@ -0,0 +1,28 @@
+  }
+
+  &__footer {
+    margin-bottom: 32px;
```
точно ли здесь нужен такой отступ?

lk4n replied:
Согласен, в лекции №2 отступ для заголовка равен 30px. 
Это была заготовка на будущее. В основном макете кроме названия карточки, есть ещё и категория. И там отступ 32px.
![card_img](https://user-images.githubusercontent.com/105793481/175349944-0e002f74-a5f2-4aad-883c-68b591b5c0e5.png)

PS. У меня вопрос по padding, margin. Когда лучше использовать `margin`, а когда `padding`? Отступ выше мог бы задать с помощью `padding`. Есть какой-то паттерн, хорошая практика, в каких случаях, что применять?

GSemikozov replied:
снизу - это все-таки паддинг - паддинги это внутренние отступы, когда надо отодвинуть элементы внутри, а марджины - это внешние, когда надо отодвинуться от других элементов / блоков

lk4n replied:
Исправил внутренний отступ на `padding`.
Изменил меню, чтобы блок `sidebar` не залезал во внутренности блока `menu`. Было в css:
`.sidebar .menu__item_current {}`
Стало:
`.menu_component .menu__item_current {}`
В html:
`class="menu menu_component sidebar__menu"`
где `sidebar__menu` отвечает только за внешнюю геометрию.

GSemikozov commented on components/card/_card.scss:
```diff
@@ -0,0 +1,28 @@
+
+  &__content {
+    flex: 1;
+    padding: 25px;
```
вроде не совпадает с макетом?)

lk4n replied:
Падинги слева и справа правильные ( в лекции 26,24; в главном макете 25).
Задумка что компонент сам выстроится посередине, как по вертикали, так и по горизонтали. Например, как аватар. 
Доработал: 

- добавил выравнивание по горизонтали
- подогнал высоту компонента, был слишком "высоким"

GSemikozov replied:
оки

GSemikozov commented on components/histogram/_histogram.scss:
```diff
@@ -0,0 +1,56 @@
+
+  &__header {
+    position: relative;
+    padding-right: 65px;
```
кажется лишним

lk4n replied:
`padding-right: 65px;` убрал

lk4n committed:
Fixes for histogram, card and main page

Changed histogram tall
Update card with category element
Fixed card inner alignment

lk4n committed:
Update card component

lk4n committed:
Update card component

Separate cosmetic and layout markup for menu block. All cosmetic styles applied using modificators and all positioning and external geometry with mixins.

lk4n committed:
Merge branch 'layout__main-page_lection-3' of github.com:lk4n/htmlcss-20220519_lk into layout__main-page_lection-3

lk4n committed:
main.css update
